frigate/log.py

```diff
@@ -1,71 +1,69 @@
-# adapted from https://medium.com/@jonathonbao/python3-logging-with-multiprocessing-f51f460b8778
+import atexit
 import logging
 import multiprocessing as mp
 import os
-import queue
-import signal
 import threading
 from collections import deque
-from logging import handlers
-from multiprocessing import Queue
-from types import FrameType
+from contextlib import AbstractContextManager, ContextDecorator
+from logging.handlers import QueueHandler, QueueListener
+from types import TracebackType
 from typing import Deque, Optional
 
-from setproctitle import setproctitle
+from typing_extensions import Self
 
 from frigate.util.builtin import clean_camera_user_pass
 
+LOG_HANDLER = logging.StreamHandler()
+LOG_HANDLER.setFormatter(
+    logging.Formatter(
+        "[%(asctime)s] %(name)-30s %(levelname)-8s: %(message)s",
+        "%Y-%m-%d %H:%M:%S",
+    )
+)
 
-def listener_configurer() -> None:
-    root = logging.getLogger()
+LOG_HANDLER.addFilter(
+    lambda record: not record.msg.startswith("You are using a scalar distance function")
```

Comment:
I am seeing an issue here with an unrelated exception that is thrown

```
  File "/usr/lib/python3.9/logging/__init__.py", line 1471, in error
    self._log(ERROR, msg, args, **kwargs)
  File "/usr/lib/python3.9/logging/__init__.py", line 1585, in _log
    self.handle(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 1595, in handle
    self.callHandlers(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 1657, in callHandlers
    hdlr.handle(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 944, in handle
    rv = self.filter(record)
  File "/usr/lib/python3.9/logging/__init__.py", line 804, in filter
    result = f(record) # assume callable - will raise if not
  File "/workspace/frigate/frigate/log.py", line 26, in <lambda>
    lambda record: not record.msg.startswith("You are using a scalar distance function")
AttributeError: 'FileNotFoundError' object has no attribute 'startswith'
```

seems like it might need to check type before assuming it is a string

Reply:
Curious that this didn't happen before. Apparently you're normally supposed to use `getMessage()` with `LogRecords`. This should be fixed now.

Reply:
Give it a test and resolve if so. I can't reproduce.